### PR TITLE
Apim 6026 save default page fix backup

### PR DESCRIPTION
**Pull request by illichso**
## Issue

https://gravitee.atlassian.net/browse/APIM-6026

## Description

Documentation Default Pages : No Confirmation message for saving/publishing the page

## Additional context

https://github.com/user-attachments/assets/9802e253-0b6d-464f-8ed2-e91fda74b30c

This is the routing for v4/documentation now:
Screenshot 2024-07-31 at 10 05 44

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-wuqbrewhrb.chromatic.com)

### CODE REVIEW

File: gravitee-apim-console-webui/src/management/api/documentation-v4/api-documentation-v4.module.ts [hunk 1]
```diff
@@ -71,19 +70,14 @@ import { GioApiMetadataListModule } from '../component/gio-api-metadata-list/gio
...
     ApiDocumentationV4EditFolderDialog,
-    ApiDocumentationV4EditPageComponent,
     ApiDocumentationV4DefaultPageComponent,
     ApiDocumentationV4AddPageButtonComponent,
     ApiDocumentationV4PagesListComponent,
-    ApiDocumentationV4FileUploadComponent,
     ApiDocumentationV4MetadataComponent,
   ],
-  exports: [ApiDocumentationV4Component],
+  exports: [ApiDocumentationV4Component /* , ApiDocumentationV4PageTitleComponent*/],
```
Comment (jourdiw) [inline gravitee-apim-console-webui/src/management/api/documentation-v4/api-documentation-v4.module.ts]: Remove comment

File: gravitee-apim-console-webui/src/management/api/documentation-v4/components/documentation-edit-page/documentation-edit-page.component.ts [hunk 2]
```diff
@@ -56,11 +77,52 @@ interface EditPageForm {
...
+  templateUrl: './documentation-edit-page.component.html',
+  styleUrl: './documentation-edit-page.component.scss',
 })
-export class ApiDocumentationV4EditPageComponent implements OnInit, OnDestroy {
+export class DocumentationEditPageComponent implements OnInit, OnDestroy {
+  @Input()
+  goBackRouterLink: string[];
+
+  @Input()
+  createHomepage: string;
```
Comment (jourdiw) [inline gravitee-apim-console-webui/src/management/api/documentation-v4/components/documentation-edit-page/documentation-edit-page.component.ts]: Shouldn't this be a boolean?

File: gravitee-apim-console-webui/src/management/api/documentation-v4/documentation-default-page/api-documentation-v4-default-page.component.ts [hunk 3]
```diff
@@ -83,7 +83,7 @@ export class ApiDocumentationV4DefaultPageComponent implements OnInit, OnDestroy
   }
 
   editPage(pageId: string) {
-    this.router.navigate([pageId], {
+    this.router.navigate(['homepage/' + pageId], {
```
Comment (jourdiw) [inline gravitee-apim-console-webui/src/management/api/documentation-v4/documentation-default-page/api-documentation-v4-default-page.component.ts]: This is fine, but just `'homepage'` works. Angular will add the `"/"` between the items

Comment (illichso) [reply gravitee-apim-console-webui/src/management/api/documentation-v4/documentation-default-page/api-documentation-v4-default-page.component.ts]: No, if I do it I get an error:
 NG04002: Cannot match any routes. URL Segment: 'default/apis/0643f7cd-bad0-4f09-83f7-cdbad06f0955/v4/documentation/default-pages/homepage5066cf3f-7817-45e7-a6cf-3f7817b5e747'

File: gravitee-apim-console-webui/src/management/api/documentation-v4/api-documentation-v4.module.ts (hunk 1 above)
Comment (illichso) [reply gravitee-apim-console-webui/src/management/api/documentation-v4/api-documentation-v4.module.ts]: Done

File: gravitee-apim-console-webui/src/management/api/documentation-v4/components/documentation-edit-page/documentation-edit-page.component.ts (hunk 2 above)
Comment (illichso) [reply gravitee-apim-console-webui/src/management/api/documentation-v4/components/documentation-edit-page/documentation-edit-page.component.ts]: Done

File: gravitee-apim-console-webui/src/management/api/documentation-v4/documentation-default-page/api-documentation-v4-default-page.component.ts (hunk 3 above)
Comment (illichso) [reply gravitee-apim-console-webui/src/management/api/documentation-v4/documentation-default-page/api-documentation-v4-default-page.component.ts]: @jourdiw can we please merge it?

File: gravitee-apim-console-webui/src/management/api/documentation-v4/api-documentation-v4.module.ts [hunk 4]
```diff
@@ -71,16 +70,11 @@ import { GioApiMetadataListModule } from '../component/gio-api-metadata-list/gio
     ApiDocumentationV4EmptyStateComponent,
     ApiDocumentationV4ListNavigationHeaderComponent,
     ApiDocumentationV4HomePageHeaderComponent,
-    ApiDocumentationV4VisibilityComponent,
-    ApiDocumentationV4ContentEditorComponent,
-    ApiDocumentationV4PageTitleComponent,
-    ApiDocumentationV4BreadcrumbComponent,
+    // ApiDocumentationV4PageTitleComponent,
```
Comment (jourdiw) [inline gravitee-apim-console-webui/src/management/api/documentation-v4/api-documentation-v4.module.ts]: Remove comment

**Review by jourdiw — APPROVED** [review]
Good to go once last part comment resolved

File: gravitee-apim-console-webui/src/management/api/documentation-v4/api-documentation-v4.module.ts (hunk 4 above)
Comment (illichso) [reply gravitee-apim-console-webui/src/management/api/documentation-v4/api-documentation-v4.module.ts]: Done

File: gravitee-apim-console-webui/src/management/api/documentation-v4/documentation-default-page/api-documentation-v4-default-page.component.ts (hunk 3 above)
Comment (jourdiw) [reply gravitee-apim-console-webui/src/management/api/documentation-v4/documentation-default-page/api-documentation-v4-default-page.component.ts]: ah I see. It should be:

```suggestion
    this.router.navigate(['homepage', pageId], {
```

Comment (illichso) [reply gravitee-apim-console-webui/src/management/api/documentation-v4/documentation-default-page/api-documentation-v4-default-page.component.ts]: Done

**Pull request merged**